### DISCUSSION TITLE
Why not support create a nfs with special crush root ?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#11195] [Q&A]
Because we rebuild the crush root. the cursh rule of "replicated_rule" is invalid, we want to ceate nfs with new cursh rule.

**Comment by zhucan** [comment]
and we don't want to create a built-in pool of ".nfs".

**Reply by travisn** [reply]
If you want to create a different root, did you try adding the `crushRoot` property to the [.nfs pool CR](https://github.com/rook/rook/blob/master/deploy/examples/nfs-test.yaml#L16-L26)?

**Reply by zhucan** [reply]
so we need to create the built-in pool firstly, and then create the nfs. if we create more nfs concurrency, we need to check the built-in pool exists.

**Comment by zhucan** [comment]
@travisn https://github.com/rook/rook/blob/0937eaee38999ddf3eaf618b5933112dc9128d1b/pkg/operator/ceph/nfs/nfs.go#L315 Please take a look at  this code, if we want to use the built-in  pool for nfs, we need to check the the pool of .nfs whether it exists, when creating nfs, it will create the pool .nfs ignore the pool whether it exists and return err, the deployment of the nfs will not be deployed.

**Reply by travisn** [reply]
Not sure I understand. You are concerned about the timing of creating the `.nfs` pool?

**Reply by zhucan** [reply]
no, Before create the nfs server,  we had created the pool of .nfs manually and the deployment of the nfs server will not be created, because it still try to create a .nfs pool and return error.

**Reply by travisn** [reply]
If the .nfs pool is created first, the nfs server should start. @BlaineEXE ?

**Reply by BlaineEXE** [reply]
Travis is correct. However, if you created the .nfs pool manually, it may have been creating with a configuration that is invalid for CephNFS.

**Reply by BlaineEXE** [reply]
It's also unclear to me whether the "return err" is an observed case or a suspected case. In all my testing, `ceph osd pool create` returns success if the pool already exists.

**Reply by zhucan** [reply]
I had tested again, you are correct. Thanks. @travisn @BlaineEXE

**Reply by zhucan** [reply]
it's more convenient that add the crushRoot to the nfs yaml like objectstore.
